File: app/src/main/java/com/seeaspark/DocActivity.java
package com.seeaspark;

import android.content.Context;
import android.graphics.Bitmap;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import org.jetbrains.annotations.NotNull;

import java.net.URLEncoder;

/**
 * Created by dev on 28/8/18.
 */

public class DocActivity extends BaseActivity {

    String docUrl, name, mPic;
    LinearLayout llOuterDoc;
    ImageView imgBack, imgProfileAvatar;
    TextView txtName;
    WebView wvDoc;
    ProgressBar pbHeaderProgress;

    @Override
    public int getContentView() {
        return R.layout.activity_doc;
    }

    @Override
    public void initUI() {
        docUrl = getIntent().getStringExtra("url");
        name = getIntent().getStringExtra("name");
        mPic = getIntent().getStringExtra("pic");

        llOuterDoc = (LinearLayout) findViewById(R.id.llOuterDoc);
        imgBack = (ImageView) findViewById(R.id.imgBack);
        imgProfileAvatar = (ImageView) findViewById(R.id.imgProfileAvatar);
        txtName = (TextView) findViewById(R.id.txtName);
        wvDoc = (WebView) findViewById(R.id.wvDoc);
        pbHeaderProgress = (ProgressBar) findViewById(R.id.pbHeaderProgress);

        txtName.setText(name);
        Picasso.with(this).load(mPic).placeholder(R.drawable.placeholder_image).into(imgProfileAvatar);
    }

    @Override
    public void displayDayMode() {
        llOuterDoc.setBackgroundColor(ContextCompat.getColor(this, R.color.white_color));
        txtName.setTextColor(ContextCompat.getColor(this, R.color.black_color));
    }

    @Override
    public void displayNightMode() {
        llOuterDoc.setBackgroundColor(ContextCompat.getColor(this, R.color.black_color));
        txtName.setTextColor(ContextCompat.getColor(this, R.color.white_color));
    }

    @Override
    public void onCreateStuff() {
        try {
            wvDoc.getSettings().setJavaScriptEnabled(true);
            wvDoc.getSettings().setAllowFileAccess(true);
            wvDoc.setScrollBarStyle(View.SCROLLBARS_INSIDE_OVERLAY);
            wvDoc.setWebViewClient(new MyBrowser());
//            docUrl = URLEncoder.encode(docUrl, "UTF-8");
            wvDoc.loadUrl("http://docs.google.com/gview?embedded=true&url=" + docUrl);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void initListener() {
        imgBack.setOnClickListener(this);
    }

    @NotNull
    @Override
    public Context getContext() {
        return this;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.imgBack:
                moveBack();
                break;
        }
    }

    private void moveBack() {
        finish();
        overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_right);
    }

    @Override
    public void onBackPressed() {
        moveBack();
    }

    private class MyBrowser extends WebViewClient {

        public MyBrowser() {

        }

        @Override
        public void onLoadResource(WebView view, String url) {

            super.onLoadResource(view, url);
        }

        @Override
        public boolean shouldOverrideUrlLoading(WebView view, String url) {
            view.loadUrl(url);
            return true;
        }

        @Override
        public void onPageStarted(WebView view, String url, Bitmap favicon) {
            super.onPageStarted(view, url, favicon);
            pbHeaderProgress.setVisibility(View.VISIBLE);
        }

        @Override
        public void onPageFinished(WebView view, String url) {
            super.onPageFinished(view, url);
            pbHeaderProgress.setVisibility(View.GONE);
        }
    }
}

File: app/src/main/java/com/seeaspark/DocActivity.kt
package com.seeaspark

import android.content.Intent
import android.graphics.Bitmap
import android.os.Build
import android.support.annotation.RequiresApi
import android.support.v4.content.ContextCompat
import android.view.View
import android.webkit.WebView
import android.webkit.WebViewClient
import kotlinx.android.synthetic.main.activity_doc.*

/**
 * Created by dev on 14/8/18.
 */
class DocActivity : BaseActivity() {

    internal var select_path = ""
    internal var name = ""
    internal var url = ""

    override fun getContentView() = R.layout.activity_doc

    override fun initUI() {
        select_path = intent.getStringExtra("select_path")
        name = intent.getStringExtra("name")
        txtName.text = name
        if(select_path.contains("http")){
            url = select_path
        }else{
            url = ""+select_path
        }
    }

    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN)
    override fun displayDayMode() {
        llOuterDoc.setBackgroundColor(ContextCompat.getColor(this, R.color.white_color))
        txtSendDoc.setTextColor(ContextCompat.getColor(this, R.color.black_color))
        txtName.setTextColor(ContextCompat.getColor(this, R.color.black_color))
    }

    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN)
    override fun displayNightMode() {
        llOuterDoc.setBackgroundColor(ContextCompat.getColor(this, R.color.black_color))
        txtSendDoc.setTextColor(ContextCompat.getColor(this, R.color.white_color))
        txtName.setTextColor(ContextCompat.getColor(this, R.color.white_color))
    }

    override fun onCreateStuff() {
        wvDoc.getSettings().setJavaScriptEnabled(true)
        wvDoc.getSettings().setAllowFileAccess(true);
        wvDoc.setScrollBarStyle(View.SCROLLBARS_INSIDE_OVERLAY)
//        wvDoc.setWebViewClient(MyBrowser())
        wvDoc.loadUrl("https://docs.cashfree.com/docs/android/guide/")
    }

    override fun initListener() {
        imgBack.setOnClickListener(this)
//        txtSend.setOnClickListener(this)
    }

    override fun getContext() = this

    override fun onClick(v: View?) {
        when (v) {
            imgBack -> {
                moveBack()
            }
//            txtSend -> {
//                sendMessage("")
//            }
        }
    }

    override fun onBackPressed() {
        moveBack()
    }

    private fun moveBack() {
        finish()
        overridePendingTransition(R.anim.slide_in_right, R.anim.slide_out_right)
    }

    internal fun sendMessage(message: String) {
        val intent = Intent()
        intent.putExtra("caption", message)
        intent.putExtra("selected_name", select_path)
        setResult(RESULT_OK, intent)
        finish()
    }


    private inner class MyBrowser : WebViewClient() {

        override fun onLoadResource(view: WebView, url: String) {

            super.onLoadResource(view, url)
        }

        override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
            view.loadUrl(url)
            return true
        }

        override fun onPageStarted(view: WebView, url: String, favicon: Bitmap) {
            super.onPageStarted(view, url, favicon)
            pbHeaderProgress.visibility = View.VISIBLE
        }

        override fun onPageFinished(view: WebView, url: String) {
            super.onPageFinished(view, url)
            pbHeaderProgress.visibility = View.GONE
        }
    }

}